stripe/src/main/java/com/stripe/android/view/CardInputView.java

```diff
@@ -40,9 +55,118 @@ private void initView() {
         inflate(getContext(), R.layout.card_input_view, this);
 
         mScrollView = (LockableHorizontalScrollView) findViewById(R.id.root_scroll_view);
-        mCardNumberEditText = (EditText) findViewById(R.id.et_card_number);
+        mCardNumberEditText = (CardNumberEditText) findViewById(R.id.et_card_number);
         mExpiryDateEditText = (EditText) findViewById(R.id.et_expiry_date);
         mCvcNumberEditText = (EditText) findViewById(R.id.et_cvc_number);
-        mCardNumberSpace = (Space) findViewById(R.id.space_in_container);
+        mCardNumberSpace = findViewById(R.id.space_in_container);
+        mCardNumberIsViewed = true;
+
+        mCardNumberEditText.setOnFocusChangeListener(new OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (hasFocus) {
+                    scrollLeft();
+                }
+            }
+        });
+
+        mExpiryDateEditText.setOnFocusChangeListener(new OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                if (hasFocus) {
```

Comment:
Realized the answer: when `mExpiryDateEditText` is `INVISIBLE`, the call to requestFocus fails, because an `INVISIBLE` item cannot receive focus. Ah, and I realized that I don't need to call to set focus here, because that's already taken care of in scrollRight()